src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+
+  const menCentury = men.filter(man =>
+    Math.ceil(man.died / 100) === century);
+
+  const MenAverageAge = men.reduce((sum, x) =>
+    sum + (x.died - x.born), 0) / men.length;
+
+  const MenAverageAgeCentury = menCentury.reduce((sum, x) =>
+    sum + (x.died - x.born), 0) / menCentury.length;
+
+  return century ? MenAverageAgeCentury : MenAverageAge;
```

Comment:
It's better to add a check for century existence at the start of function and delete redundant second reduce method:
```suggestion
  men = century
  	? men.filter(man => Math.ceil(man.died / 100) === century)
  	: men;

  const menAverageAge = men.reduce((sum, x) =>
    sum + (x.died - x.born), 0) / men.length;

  return menAverageAge;
```